breacharbiter.go

```diff
@@ -30,6 +30,51 @@ import (
 // continue from the persisted state.
 var retributionBucket = []byte("retribution")
 
+// BreachConfig bundles the required subsystems used by the breach arbiter. An
+// instance of BreachConfig is passed to newBreachArbiter during instantiation.
+type BreachConfig struct {
+	// Signer is used by the breach arbiter to generate sweep transactions,
+	// which move coins from previously open channels back to the user's
+	// wallet.
+	Signer lnwallet.Signer
+
+	// DB provides access to the user's channels, allowing the breach
+	// arbiter to determine the current state of a user's channels, and how
+	// it should respond to channel closure.
+	DB *channeldb.DB
+
+	// Wallet facilitates sweep transaction publication and further
+	// facilitates the generation of receiver scripts when sweeping funds.
+	Wallet *lnwallet.LightningWallet
+
+	// Notifier provides a publish/subscribe interface for event driven
+	// notifications regarding the confirmation of txids.
+	Notifier chainntnfs.ChainNotifier
+
+	// ChainIO is used by the breach arbiter to determine the current height
+	// of the blockchain, which is required to subscribe for spend
+	// notifications from Notifier.
+	ChainIO lnwallet.BlockChainIO
+
+	// Estimator is used by the breach arbiter to determine an appropriate
+	// fee level when generating, signing, and broadcasting sweep
+	// transactions.
+	Estimator lnwallet.FeeEstimator
+
+	// HtlcSwitch gives the breach arbiter access to shutdown any channel
+	// links for which it detects a breach, ensuring no further activity
+	// will continue across the link.
+	HtlcSwitch *htlcswitch.Switch
```

Comment:
We don't need the entire switch. Instead, we can simply take something with identical arguments to `CloseLink`.